bounties/level-2/stellar-accounts-as-a-service.md

```diff
@@ -0,0 +1,190 @@
+# Stellar accounts as a service
+
+## Award
+| Hunter | Reviewer | Type
+| :- | :- | :-
+| 2000 XLM | 800 XLM | Continuous
+
+📜&nbsp; View [existing submissions](https://github.com/tyvdh/stellar-quest-bounties/issues?q=is%3Aissue+stellar-accounts-as-a-service+) for this bounty. \
+🔵&nbsp; Start [hunting](https://github.com/tyvdh/stellar-quest-bounties/issues/new?assignees=&labels=&template=begin-the-hunt.yml&title=%F0%9F%94%B5+%60stellar-accounts-as-a-service.md%60) this bounty.
+
+## Description
+
+### What is this task?
+
+Each Stellar account needs to hold 1 XLM to be valid. This can be costly when companies like exchanges want to build
+on the Stellar network.
+
+Currently, exchanges assign each of their users a unique id that should be put in the memo field of a transaction when depositing.
+This approach is error-prone. Users can forget to add the memo, or they can input the wrong memo.
+
+To solve this issue [muxed accounts](https://github.com/stellar/stellar-protocol/blob/master/core/cap-0027.md) were added to Stellar. Muxed accounts are essentially a way to combine Stellar addresses with integer identifiers.
+Payments sent to a muxed account are actually sent to the account used to create the muxed account. However, the muxed account identifier is also added to the transaction.
+
+For this bounty the task is to create an API for a [custodial](https://www.gemini.com/cryptopedia/crypto-wallets-custodial-vs-noncustodial) wallet service using muxed accounts.
+A simple database should be used to keep track of the registered users, their muxed account addresses and their balances.
+
+The api should have the following endpoints:
+
+```
+openapi: 3.0.1
+info:
+  title: Stellar accounts as a service
+  description: 'This is the api specification for the Stellar accounts as a service bounty.'
+  version: 1.0.0
+tags:
+- name: auth
+  description: Authenticate yourself
+- name: info
+  description: Info
+- name: pay
+  description: Different ways to pay
+paths:
+  /register:
+    post:
+      tags:
+      - auth
+      summary: Create an account.
+      requestBody:
+        description: Credentials to register with.
+        content:
+          application/json:
+            schema:
+              properties:
+                username:
+                  type: string
+                password:
+                  type: string
+        required: true
+      responses:
```

Comment:
Maybe a distinct response for when the desired username already is signed up? [409 CONFLICT](https://httpstatuses.com/409) may be appropriate.

Reply:
This has been added in a new commit.